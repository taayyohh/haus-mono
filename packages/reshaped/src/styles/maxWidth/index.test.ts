import getMaxWidthStyles from "./index";

describe("Styles/MaxWidth", () => {
	test("handles px value", () => {
		expect(getMaxWidthStyles("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(getMaxWidthStyles("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(getMaxWidthStyles(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getMaxWidthStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getMaxWidthStyles({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
