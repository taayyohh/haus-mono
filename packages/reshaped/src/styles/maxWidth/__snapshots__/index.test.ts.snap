// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/MaxWidth > handles % value 1`] = `
{
  "classNames": [
    "_literal_c65e37",
  ],
  "variables": {
    "--rs-max-w-s": "50%",
  },
}
`;

exports[`Styles/MaxWidth > handles px value 1`] = `
{
  "classNames": [
    "_literal_c65e37",
  ],
  "variables": {
    "--rs-max-w-s": "50px",
  },
}
`;

exports[`Styles/MaxWidth > handles responsive value 1`] = `
{
  "classNames": [
    "_literal_c65e37",
    "_literal--l_c65e37",
  ],
  "variables": {
    "--rs-max-w-l": "50%",
    "--rs-max-w-s": "50px",
  },
}
`;

exports[`Styles/MaxWidth > handles undefined value 1`] = `null`;

exports[`Styles/MaxWidth > handles unit value 1`] = `
{
  "classNames": [
    "_unit_c65e37",
  ],
  "variables": {
    "--rs-max-w-s": 5,
  },
}
`;
